Update SafeMutex to implement IDisposable

scottpidzarko opened the pull request:
The wrapped Mutex / CMutex class in SafeMutex is IDisposable, therefore SafeMutex should follow suit.

scottpidzarko committed:
Update SafeMutex.cs to implement IDisposable

scottpidzarko committed:
Update SafeMutex.cs to use correct class name in boilerplate commented out finalizer

astingen commented:
Can you remove the commented-out code, and change this to merge into dev instead of master?
Thanks!

scottpidzarko committed:
chore: Update SafeMutex.cs to remove commented out boilerplate

IDisposable boilerplate generates commented out finalizer. Removing commented out finalizer at request of upstream maintainer.

scottpidzarko commented:
Done
